google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITJsonWriteReadTest.java

```diff
@@ -0,0 +1,156 @@
+            "CREATE TABLE "
+                + TABLE_NAME
+                + "("
+                + "Id INT64 NOT NULL,"
+                + "json JSON"
+                + ") PRIMARY KEY (Id)");
+    databaseClient = testHelper.getDatabaseClient(database);
+  }
+
+  @Test
+  public void testWriteValidJsonValues() throws IOException {
+    assumeFalse("Emulator does not yet support JSON", EmulatorSpannerHelper.isUsingEmulator());
+
+    List<String> resources = getJsonFilePaths(RESOURCES_DIR + File.separator + VALID_JSON_DIR);
+
+    long id = 0L;
+    List<Mutation> mutations = new ArrayList<>();
+    List<String> jsons = new ArrayList<>();
+    for (String resource : resources) {
+      String jsonStr =
+          Resources.toString(
+              Resources.getResource(this.getClass(), VALID_JSON_DIR + File.separator + resource),
+              StandardCharsets.UTF_8);
+      jsons.add(jsonStr);
+      Mutation mutation =
+          Mutation.newInsertBuilder(TABLE_NAME)
+              .set("Id")
+              .to(id++)
+              .set("json")
+              .to(Value.json(jsonStr))
+              .build();
+      mutations.add(mutation);
+    }
+
+    databaseClient.write(mutations);
+    ResultSet resultSet =
+        databaseClient.singleUse().executeQuery(Statement.of("SELECT * FROM " + TABLE_NAME));
+    int count = 0;
+    while (resultSet.next()) {
+      count++;
+    }
+    assertEquals(count, resources.size());
+  }
+
+  @Test
+  public void testWriteAndReadInvalidJsonValues() throws IOException {
+    assumeFalse("Emulator does not yet support JSON", EmulatorSpannerHelper.isUsingEmulator());
+
+    List<String> resources = getJsonFilePaths(RESOURCES_DIR + File.separator + INVALID_JSON_DIR);
+
+    for (String resource : resources) {
+      String jsonStr =
+          Resources.toString(
+              Resources.getResource(this.getClass(), INVALID_JSON_DIR + File.separator + resource),
+              StandardCharsets.UTF_8);
+
+      try {
+        databaseClient.write(
+            Collections.singletonList(
+                Mutation.newInsertBuilder(TABLE_NAME)
+                    .set("Id")
+                    .to(100L)
+                    .set("json")
+                    .to(Value.json(jsonStr))
+                    .build()));
+        fail(resource + " should be rejected.");
+      } catch (Exception e) {
+        System.out.println("Failed to insert JSON: " + jsonStr + ". From: " + resource);
+        e.printStackTrace();
+      }
+    }
+  }
+
+  private List<String> getJsonFilePaths(String folder) {
+    String fixturesRoot = Resources.getResource(folder).getPath();
+    final Path fixturesRootPath = Paths.get(fixturesRoot);
+    try {
+      return Files.walk(fixturesRootPath)
+          .filter(Files::isRegularFile)
+          .map(path -> fixturesRootPath.relativize(path).toString())
+          .filter(x -> x.endsWith(".json"))
+          .collect(Collectors.toList());
+    } catch (IOException e) {
```

Comment:
nit: I don't think you need the try-catch here. The test methods already declare `throws IOException`, so we could just as well just let that bubble up here.